src/theme/DocItem/Metadata/determineCanonical.js

```diff
@@ -107,7 +107,7 @@ function determineCanonicalFromDoc(currentDoc, currentPlugin) {
     .find((doc) => doc.id === unversionedId);
 
   if (match) {
-    if (/(optimize|docs)((next|[0-9\.]*)\/)/.test(match.path)) {
+    if (/(optimize|docs)\/((next|[0-9\.]+)\/)/.test(match.path)) {
```

Comment:
The old regex was matching _every single page_, because it was only requiring the `optimize|docs` at the beginning